Sources/IndexStoreDB/IndexStoreDB.swift

```diff
@@ -275,6 +285,41 @@ public final class IndexStoreDB {
     return result
   }
     
+  public func symbolProvider(for sourceFilePath: String) -> SymbolProviderKind? {
+    var result: SymbolProviderKind? = nil
+    indexstoredb_index_units_containing_file(impl, sourceFilePath) { unit in
+      let providerKind: SymbolProviderKind? = switch indexstoredb_unit_info_symbol_provider_kind(unit) {
+      case INDEXSTOREDB_SYMBOL_PROVIDER_KIND_SWIFT:
+        .swift
+      case INDEXSTOREDB_SYMBOL_PROVIDER_KIND_CLANG:
+        .clang
+      case INDEXSTOREDB_SYMBOL_PROVIDER_KIND_UNKNOWN:
+        nil
+      default:
+        preconditionFailure("Unknown enum case in indexstoredb_symbol_provider_kind_t")
+      }
+
+      let mainFilePath = String(cString: indexstoredb_unit_info_main_file_path(unit))
+      if providerKind == .swift && mainFilePath != sourceFilePath {
+        // We have a unit that is "included" from Swift. This happens for header
+        // files that Swift files depend on. But Swift doesn't have includes and
+        // we shouldn't infer the header file's language to Swift based on this unit.
+        // Ignore it.
+        return true
+      }
+
+      if result == nil {
```

Comment:
Is this worth it? Would be a pretty weird state to be in, may as well just return the first.

Reply:
I don’t have strong opinions either way. `indexstoredb_index_units_containing_file` also shouldn’t return many files so not sure if exiting early really makes a difference.